src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,505 @@ title: Builder.io & Astro
+
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key to Astro, create a `.env` file in the root of your project (if one does not already exist) and add the following variable:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+```
+
+Now, you should be able to use this API key in your project.
+
+:::note
+At the time of writing, [this key is public](https://www.builder.io/c/docs/using-your-api-key#prerequisites), so you don't have to worry about hiding or encrypting it.
+:::
+
+If you would like to have IntelliSense for your environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly BUILDER_API_PUBLIC_KEY: string;
+}
+```
+
+Your project should now include these files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+
+## Making a blog with Astro and Builder
+
+With the setup above, you can now create a blog with Astro and Builder.
+
+### Creating a model for a blog post
+
+
+In the Builder app create the model that will represent a blog post: go to the **Models** tab and click the **Create Model** button to create a model with the following fields and values:
+
+- **Type:** Section
+- **Name:** "blogpost"
+- **Description:** "This model is for a blog post"
+
+:::note[Why a section model?]
+In a section model, the content can sit between other elements of a page - like a `<header>` and `<footer>`, for example.
+:::
+
+In your new model use the **New Custom Field** button to create 2 new fields:
+
+1. Text field
+    - **Name:** "title"
+    - **Required:** Yes
+    - **Default value:** "I forgot to give this a title"
+
+    (leave the other parameters as their defaults)
+
+2. Text field
+    - **Name:** "slug"
+    - **Required:** Yes
+    - **Default value:** "some-slugs-take-their-time"
+
+    (leave the other parameters as their defaults)
+
+Then click the **Save** button in the upper right. 
+
+And put the name of the model in the `.env` file for convenience:
+```ini title=".env" ins={2}
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+BUILDER_BLOGPOST_MODEL='blogpost'
+```
+
+:::caution
+There are some pitfalls with the `slug` field:
+
+* Make sure your slug is not just a number. This seems to break the fetch request to Builder's API. 
+* Make sure your slugs are unique since your site's routing will depend on that.
+:::
```

Comment:
```suggestion
The instructions below create an Astro blog using a Builder model (Type: "Section") called `blogpost` that contains two required text fields: `title` and `slug`.
```
We can figure out exactly how to say this, but something like this is my idea for cutting this section, and referring to this in the prereq section above.